File: Promedios.py
def obtener_num(mensaje, minimo=1, maximo=None):  # Función para obtener valores enteros válidos
    valido = False
    while not valido:
        entrada = input(mensaje)
        if entrada.isdigit():
            valor = int(entrada)
            if valor >= minimo and (maximo is None or valor <= maximo):
                return valor
            else:
                if maximo is not None:
                    print(f"Error: El número debe estar entre {minimo} y {maximo}")
                else:
                    print(f"Error: El número debe ser mayor o igual a {minimo}")
        else:
            print("Error: Ingrese un número entero válido")

def calcular_promedio(lista):
    suma = 0
    for i in range(len(lista)):
        suma += lista[i]
    promedio = suma / len(lista)
    return promedio

def contar_aprobados_reprobados(lista_estudiantes):
    aprobados = 0
    reprobados = 0
    for est in lista_estudiantes:
        if est["estado"] == "Aprobado":
            aprobados += 1
        else:
            reprobados += 1
    return aprobados, reprobados

# Programa principal
estudiantes = []
print("Sistema de promedios de estudiantes\n")
num_estudiantes = obtener_num("\nIngrese el número de estudiantes: ")
print("Ingrese las calificaciones de los estudiantes:")

for i in range(num_estudiantes):
    print(f"\nEstudiante {i+1}")
    
    # Validación nombre no vacío
    while True:
        nombre = input("Nombre: ")
        if nombre.strip() == "":
            print("Error: El nombre no puede estar vacío")
        else:
            break

    calificaciones = []  # Reinicia la lista de calificaciones por estudiante    
    
    for j in range(3):
        calificacion = obtener_num(f"Calificación Materia {j + 1} (0-10): ", 0, 10)
        calificaciones.append(calificacion)
    
    promedio = calcular_promedio(calificaciones)

    estudiantes.append({
        "nombre": nombre,
        "promedio": promedio,
        "estado": "Aprobado" if promedio >= 6 else "Reprobado"
    })

# Mostrar resultados
print("\n--- RESULTADOS ---")

for estudiante in estudiantes:
    print(f"{estudiante['nombre']} - Promedio: {estudiante['promedio']:.1f} - {estudiante['estado']}")

# Contar aprobados y reprobados
aprobados, reprobados = contar_aprobados_reprobados(estudiantes)

print(f"\nTotal de estudiantes aprobados: {aprobados}")
print(f"Total de estudiantes reprobados: {reprobados}")

File: Promedios.cpp
#include <iostream>
#include <string>

using namespace std;

struct Estudiante {
    string nombre;
    float promedio;
    string estado;
    int calificaciones[3];
};

int obtener_num(string mensaje, int minimo = 1, int maximo = -1) {
    string entrada;
    int valor;
    
    while(true) {
        cout << mensaje;
        getline(cin, entrada);
        
        bool es_numero = true;
        for(char c : entrada) {
            if(!isdigit(c)) {
                es_numero = false;
                break;
            }
        }
        
        if(es_numero) {
            valor = stoi(entrada);
            if(valor >= minimo && (maximo == -1 || valor <= maximo)) {
                return valor;
            } else {
                if(maximo != -1) {
                    cout << "Error: Debe estar entre " << minimo << " y " << maximo << endl;
                } else {
                    cout << "Error: Debe ser mayor o igual a " << minimo << endl;
                }
            }
        } else {
            cout << "Error: Ingrese un número entero válido" << endl;
        }
    }
}

float calcular_promedio(int calificaciones[3]) {
    float suma = 0;
    for(int i = 0; i < 3; i++) {
        suma += static_cast<float>(calificaciones[i]);
    }
    return suma / 3;
}

int main() {
    Estudiante estudiantes[100];
    int num_estudiantes;
    
    cout << "Sistema de promedios de estudiantes\n" << endl;
    num_estudiantes = obtener_num("\nIngrese el numero de estudiantes: ");
    
    for(int i = 0; i < num_estudiantes; i++) {
        cout << "\nEstudiante " << i+1 << endl;
        
        //Validar nombre
        while(true) {
            cout << "Nombre: ";
            getline(cin, estudiantes[i].nombre);
            
            string nombre_sin_espacios;
            for(char c : estudiantes[i].nombre) {
                if(c != ' ' && c != '\t' && c != '\n') {
                    nombre_sin_espacios += c;
                }
            }
            
            if(!nombre_sin_espacios.empty()) {
                estudiantes[i].nombre = nombre_sin_espacios;
                break;
            }
            cout << "Error: El nombre no puede estar vacio" << endl;
        }
        
        //OBTENER CALIFICACIONES
        for(int j = 0; j < 3; j++) {
            estudiantes[i].calificaciones[j] = obtener_num(
                "Calificacion Materia " + to_string(j+1) + " (0-10): ", 0, 10
            );
        }
        
        //CALCULAR PROMEDIO Y ESTADO
        estudiantes[i].promedio = calcular_promedio(estudiantes[i].calificaciones);
        estudiantes[i].estado = (estudiantes[i].promedio >= 6) ? "Aprobado" : "Reprobado";
    }
    
    cout << "\n--- RESULTADOS ---" << endl;
    int aprobados = 0, reprobados = 0;
    
    for(int i = 0; i < num_estudiantes; i++) {
        cout << estudiantes[i].nombre << " - Promedio: " << estudiantes[i].promedio;
        cout << " - " << estudiantes[i].estado << endl;
        
        if(estudiantes[i].estado == "Aprobado") {
            aprobados++;
        } else {
            reprobados++;
        }
    }
    
    cout << "\nTotal de estudiantes aprobados: " << aprobados << endl;
    cout << "Total de estudiantes reprobados: " << reprobados << endl;
    
    return 0;
}